### PR TITLE
[QNN EP] Support HardSigmoid

### PR DESCRIPTION
**Pull request by adrianlizarraga**
### Description
- Adds support for float32/float16 HardSigmoid on HTP backend. Decomposes `HardSigmoid(X)` into `max(0, min(1, alpha * X + beta))`.
- Fuses the sequence `X * HardSigmoid<alpha=1/6, beta=0.5>(X)` into a single `HardSwish(x)`. Only applies to non-quantized HardSigmoid/Mul.

### Motivation and Context
QNN does not natively support HardSigmoid. These changes expand model support on QNN EP.

### CODE REVIEW

**Review by jywu-msft — APPROVED** [review]

File: onnxruntime/core/providers/qnn/builder/qnn_def.h [hunk 1]
```diff
@@ -213,6 +213,22 @@ class QnnTensorWrapper {
...
+      std::swap(tensor_name_, other.tensor_name_);
+      std::swap(dimensions_, other.dimensions_);
+      std::swap(client_buf_, other.client_buf_);
+      std::swap(quant_params_, other.quant_params_);
+      std::swap(qnn_tensor_, other.qnn_tensor_);
+      SetQnnTensorName(qnn_tensor_, tensor_name_.c_str());
+      SetQnnTensorDim(qnn_tensor_, dimensions_);
+      SetQnnTensorClientBuf(qnn_tensor_, client_buf_);
+      SetQnnTensorQParams(qnn_tensor_, quant_params_.Get());
+    }
```
Comment (HectorSVC) [inline onnxruntime/core/providers/qnn/builder/qnn_def.h:227]: move to a private method to share with QnnTensorWrapper(QnnTensorWrapper&& other)

File: onnxruntime/core/providers/qnn/qnn_execution_provider.cc [hunk 2]
```diff
@@ -445,50 +450,45 @@ QNNExecutionProvider::GetSupportedNodes(const GraphViewer& graph_viewer,
...
-      LOGS(logger, WARNING) << "Failed to convert DQ -> Q sequence to QNN Convert. "
-                            << "Type: " << node_unit->OpType() << ", Node name: " << node_unit->Name() << ", "
-                            << "Message: " << convert_result.status.ErrorMessage();
-    }
-
-    bool supported = false;
+    // Try to see if this node unit can be fused.
+    std::vector<const NodeUnit*> fused_nodes;
+    Status fusion_status = TryFusions(fused_nodes, qnn_model_wrapper, *node_unit, node_unit_map,
+                                      handled_node_units, logger, false);
```
Comment (HectorSVC) [inline onnxruntime/core/providers/qnn/qnn_execution_provider.cc:460]: >false [](http://example.com/codeflow?start=66&length=5)

why do we still need this flag for op_validation if it's always false?

Comment (adrianlizarraga) [reply onnxruntime/core/providers/qnn/qnn_execution_provider.cc:460]: This should be true for the validation pass (dont want to actually create qnn node). Thanks for catching this. I'll make the change.

File: onnxruntime/core/providers/qnn/builder/qnn_fusions.cc [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+  // If validation fails here, we return Status::OK() to allow QNN EP to use the normal OpBuilder workflow.
+  QNN_RETURN_OK_IF_ERROR(qnn_model_wrapper.MakeTensorWrapper(input_def, input_tensor), logger);
+  QNN_RETURN_OK_IF_ERROR(qnn_model_wrapper.MakeTensorWrapper(output_def, output_tensor), logger);
+  QNN_RETURN_OK_IF_ERROR(qnn_model_wrapper.ValidateQnnNode(node_name,
+                                                           QNN_OP_PACKAGE_NAME_QTI_AISW,
+                                                           QNN_OP_CONVERT,
+                                                           {input_tensor.GetQnnTensor()},
+                                                           {output_tensor.GetQnnTensor()},
+                                                           {}),
+                         logger);
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/qnn_fusions.cc:112]: I want to draw attention to the above lines. These fusions call QNN's validator api before deciding to actually do the fusion. This happens even when compiling the graph.

The validation does not modify any state, and if it fails, we return Status::OK() to allow the EP to use the normal OpBuilder workflow.

**Review by HectorSVC — APPROVED** [review]
:shipit:

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**